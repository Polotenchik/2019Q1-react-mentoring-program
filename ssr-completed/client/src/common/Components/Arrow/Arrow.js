import React from 'react';
import styled, { css } from 'styled-components';

const Arrow = ({ ...props }) => (
  <ArrowWrapper {...props}>
    <i className="fas fa-arrow-up" />
  </ArrowWrapper>
);

const ArrowWrapper = styled.span`

    .fas {

        margin-left: 20px;

        ${props => props.down && css`
            transform: rotate(180deg);
        `}

         ${props => props.left && css`
            transform: rotate(-90deg);
            margin-left: 0;
         `}

         ${props => props.right && css`
            transform: rotate(90deg);
         `}

     }
`;

export default Arrow;
