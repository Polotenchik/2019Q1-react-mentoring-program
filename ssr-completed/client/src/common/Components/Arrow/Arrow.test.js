import React from 'react';
import { mount} from 'enzyme';
import Arrow from './Arrow';

describe('Arrow Snapshot', () => {
    it('Arrow Common Component should render correctly with no props', () => {
        const component = mount(
            <Arrow modifiers={['down']}/>
        );
        expect(component).toMatchSnapshot();
    });
});