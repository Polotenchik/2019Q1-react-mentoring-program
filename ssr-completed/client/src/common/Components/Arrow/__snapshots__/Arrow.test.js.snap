// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow Snapshot Arrow Common Component should render correctly with no props 1`] = `
<Arrow
  modifiers={
    Array [
      "down",
    ]
  }
>
  <styled.span
    modifiers={
      Array [
        "down",
      ]
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "ksPSpM",
            "rules": Array [
              "

    .fas {

        margin-left: 20px;

        ",
              [Function],
              "

         ",
              [Function],
              "

         ",
              [Function],
              "

     }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      modifiers={
        Array [
          "down",
        ]
      }
    >
      <span
        className="sc-bdVaJa ksPSpM"
      >
        <i
          className="fas fa-arrow-up"
        />
      </span>
    </StyledComponent>
  </styled.span>
</Arrow>
`;
